G1ANT.Addon.Selenium/Api/SeleniumDriverResourceDescription.cs

```diff
@@ -1,84 +1,118 @@
+
+        private void UpdateDriverFromResource(string filename, byte[] resourceData, string unpackFolder)
         {
-            return !DoesFileExist(unpackFolder, filename) || !AreFilesOfTheSameLength(resourceData.Length, unpackFolder, filename);
+            KillWorkingProcess(Path.GetFileNameWithoutExtension(filename), unpackFolder);
+            var startTickCount = Environment.TickCount;
+            while (!CreateDriverFileFromResoure(filename, resourceData, unpackFolder)
+                && !TimeoutOccured(startTickCount, createFileTimeout))
+                ;
+
+            if (TimeoutOccured(startTickCount, createFileTimeout))
+                throw new Exception($"Unable to unpack driver {filename}. Driver is being used by another process.");
         }
 
-        private void UpdateDriverFromResource(string filename, byte[] resourceData, string unpackFolder)
+        private bool TimeoutOccured(int startTime, int timeout) 
+            => Environment.TickCount - startTime > timeout;
+
+        private bool CreateDriverFileFromResoure(string filename, byte[] resourceData, string unpackFolder)
         {
-            KillWorkingProcess(Path.GetFileNameWithoutExtension(filename));
-            using (FileStream stream = File.Create(Path.Combine(unpackFolder, filename)))
+            try
             {
-                stream.Write(resourceData, 0, resourceData.Length);
+                using (FileStream stream = File.Create(Path.Combine(unpackFolder, filename)))
+                    stream.Write(resourceData, 0, resourceData.Length);
+
+                return true;
+            }
+            catch
+            {
+
+                return false;
             }
         }
 
-        private void KillWorkingProcess(string processName)
+        private void KillWorkingProcess(string processName, string unpackFolder)
         {
             foreach (Process proc in Process.GetProcessesByName(processName))
-            {
-                try
-                {
-                    proc.Kill();
-                }
-                catch
-                {
-
-                }
-            }
+                if (proc.MainModule.FileName.Contains(unpackFolder))
+                    KillProcessAndChildren(proc.Id);
         }
 
         private bool DoesFileExist(string folder, string fileName)
-        {
-            return File.Exists(Path.Combine(folder, fileName));
-        }
+            => File.Exists(Path.Combine(folder, fileName));
+
 
         private bool AreFilesOfTheSameLength(int length, string folder, string fileName)
+            => length == new FileInfo(Path.Combine(folder, fileName)).Length;
+
+
+        private static void KillProcessAndChildren(int pid)
         {
-            return length == new FileInfo(Path.Combine(folder, fileName)).Length;
+            if (pid == 0)
+                return;
+
+            ManagementObjectSearcher searcher = new ManagementObjectSearcher
```

Comment:
I have no idea how WIM methods behave in standard user context (without admin privileges). It has to be tested